### PR TITLE
Remove id to name and params

### PR DESCRIPTION
**Pull request by stephenpascoe**
This PR removes the problematic function luigi.task.id_to_name_and_params().  It was being used extensively when the scheduler called TaskHistory methods.  Now these methods take a luigi.scheduler.Task() instance instead so that deserialisation of task_id is unnecessary.

I have added a test which would catch #1270, #1171 and #1132.

I have also optimised an SqlAlchemy query in DbTaskHistory().

### CODE REVIEW

**Comment by erikbern** [comment]
This is great! The function has been deprecated long enough so I'm glad we're removing it. Awesome.

I think the test failure is just flakiness so I retriggered it.

File: luigi/task.py [hunk 1]
```diff
@@ -58,12 +58,6 @@ class Task2(luigi.Task):
     Register._default_namespace = namespace
 
 
-def id_to_name_and_params(task_id):
```
Comment (Tarrasch) [inline luigi/task.py]: \o/

File: luigi/tools/parse_task.py [hunk 2]
```diff
@@ -1,68 +0,0 @@
...
-# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-# License for the specific language governing permissions and limitations under
-# the License.
-
-import warnings
-
-import pyparsing as pp
-
-
-def id_to_name_and_params(task_id):
```
Comment (Tarrasch) [inline luigi/tools/parse_task.py]: DIE DIE DIE DIE!!!!!!!!!!!!!!!!!!!!!!

**Comment by Tarrasch** [comment]
LGTM. But it seems like the first 3 commits could be squashed to keep history to only logical changes.

**Comment by stephenpascoe** [comment]
I squashed the commits and now Travis is failing again.  Looks like spurious failures.

**Comment by Tarrasch** [comment]
Hmm...

selection_091

Well, it's not spurious. But something have made all tests fail. I just saw it in #1310 too. @ulzha, can you take a look? :)

**Comment by erikbern** [comment]
the test breakages have something to do with elastic search so it's kind of weird

**Comment by Tarrasch** [comment]
Since `id_to_name_and_params` will be shown in readthedocs (doh). I think this deserves merging before 2.0.

**Comment by erikbern** [comment]
huh is it visible in the docs? if they are then we should remove that too

**Comment by Tarrasch** [comment]
Every method that doesn't start with a `_` will be in the docs I think.

**Pull request merged**